enhancements/network/configure-ovs-alternative.md

```diff
@@ -0,0 +1,407 @@
+
+### API Extensions
+
+This will not directly modify the API of OpenShift. The API for this feature is
+implemented in Kubernetes-NMState, and this does not require any specific
+modifications in that project at this time.
+
+### Topology Considerations
+#### Hypershift / Hosted Control Planes
+
+This should not have any hypershift-specific considerations.
+
+#### Standalone Clusters
+
+This is primarily directed at standalone clusters, specifically baremetal and
+other on-prem environments.
+
+#### Single-node Deployments or MicroShift
+
+This does not introduce any new runtime components so should not affect
+resource consumption in SNO and MicroShift deployments.
+
+### Implementation Details/Notes/Constraints [optional]
+
+Scaleout in this implementation (using static IPs on baremetal to demonstrate
+as many parts of the process as possible) would look something like this:
+
+* Add the new node's configuration to the machine-config for its role.
+  * `oc edit mc 10-br-ex-worker`
+* Create and apply a secret containing the minimal static IP configuration
+  needed for the node to pull ignition.
+  * `oc apply -f ./extraworker-secret.yaml`
+* Create and apply a BareMetalHost resource with the network secret attached
+  via the `preprovisioningNetworkDataName` field.
+```yaml
+apiVersion: metal3.io/v1alpha1
+kind: BareMetalHost
+spec:
+  # ...
+  preprovisioningNetworkDataName: ostest-extraworker-0-network-config-secret
+```
+* Scale out the machineset to deploy the new node.
+  * `oc project openshift-machine-api`
+    `oc get machinesets`
+    `oc scale machineset [name of machineset] --replicas=[number of nodes desired after scaling]`
+
+The node will initially boot with the network configuration specified in
+`preprovisioningNetworkDataName` and then have the configuration in the
+machine-config applied.
+
+Only the first step of the process is new for this feature. The rest is
+standard baremetal scaleout when using static IPs.
+
+### Risks and Mitigations
+
+Because this gives the deployer a great deal of control over the bridge
+configuration for OVNK, they will have the ability to configure it incorrectly,
+which may cause problems either on initial deployment or at any point after.
+We will want to ensure that must-gather is collecting enough information about
+the bridge configuration for us to determine whether it is compliant with our
+guidelines, and we must have clear instructions on what is required for br-ex.
+
+Note, however, that if an incorrect bridge configuration is applied NMState
+will usually catch that and roll back the changes.
+
+### Drawbacks
+
+OpenShift has historically avoided per-host configuration for cluster nodes.
+This is a large departure from that philosophy, but that is unavoidable.
+Certain aspects of networking (notably static IPs) are by nature host-specific
+and cannot be handled any other way.
+
+Using a machine-config to do initial deployment and a
```

Comment:
How can existing nodes move off configure-ovs to this new approach?

Do cluster admins have to create the `10-br-ex-worker` machine-config (will reboot potentially all nodes) with the desired new network configuration (and be able to further update it on day 2 via k-nmstate)?

Apologies if it's stated somewhere in the doc. In any case, we would want to document such a procedure on the OCP documentation page.

Reply:
I've added a discussion of how I expect migration to work and opened a story on the epic to track verification that this works as intended: https://issues.redhat.com/browse/OPNET-508